drivers/display/gd7965.c

```diff
@@ -94,12 +97,13 @@ static inline int gd7965_write_cmd(struct gd7965_data *driver,
 
 static inline void gd7965_busy_wait(struct gd7965_data *driver)
 {
-	u32_t val = 0U;
+	int pin = gpio_pin_get(driver->busy, GD7965_BUSY_PIN);
 
-	gpio_pin_read(driver->busy, GD7965_BUSY_PIN, &val);
-	while (val == 0) {
+	while (pin > 0) {
+		__ASSERT(pin >= 0, "Failed to get pin level");
```

Comment:
We have such usage of ASSERT in tree already, e.g. `drivers/display/ssd16xx.c` but this is actually not ideal. If the busy pin is connected to GPIO expander, e.g. sx1509b the `gpio_pin_get` function can return a real run time error, e.g. due to problems communication over i2c bus. We should handle the error without the assert. On the other hand if signal on GD7965_BUSY_PIN changes fast enough that it's not feasible to handle it via a GPIO expander adding error handling is an overkill. Maybe just a comment? E.g. /* Ignoring error codes, supported by SoC GPIO modules only */.

Reply:
I am for "handle the error without the assert" but suggest to fix it in gd7965 and in ssd16xx later.